simulator/src/data/backupCircuit.js

```diff
@@ -56,7 +56,7 @@ export function backUp(scope = globalScope) {
 
 export function scheduleBackup(scope = globalScope) {
     var backup = JSON.stringify(backUp(scope));
-    if (scope.backups.length === 0 || scope.backups[scope.backups.length - 1] !== backup) {
+    if ((scope.backups.length === 0 || scope.backups[scope.backups.length - 1] !== backup) && backUp(scope).allNodes.length !== 0 ) {
```

Comment:
This change will actually not fix anything. It does fix the condition that if the undo results in empty canvas, then backup does not happen. But say if I load a saved circuit, then the max undos(till backups array is empty) will set the canvas in a nonempty state and still the history array will be overwritten